### ISSUE TITLE
2.4.11 Focus Appearance requires contrast for aliased pixels

### ISSUE THREAD

**Issue by WilcoFiers** [w3c/wcag#1348]
The definition of "Focus indication area" includes all pixels that change, that includes the aliased pixels too. It seems unreasonable to expect the aliased pixels to also have a contrast ratio of 3:1 to the unfocused state, or to the background.

**Comment by alastc** [comment]
Hi @WilcoFiers,

Ideally, we'd like to essentially ignore anti-aliasing, but account for intentional gradients.

E.g. if someone defines a 2px outline that should be sufficient regardless of anti-aliasing.

We have covered the [gradient case](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance-minimum.html#focus-indicator-box-shadow
), we could add a line to the [minimum area section](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance-minimum.html#size-of-indicator) if that helps?

There is another issue about dotted/dashed lines that is along a similar line, perhaps we can say something like:

> Where a focus indicator is defined in code as a certain size, e.g. 2px thickness, anti-aliasing can be ignored for the purposes of calculation. Dotted or dashed outlines have various levels of gaps depending on the browser, screen density, and thickness. For example, in most browsers a "dotted" line will have roughly half the number of pixels due to gaps. However, in some sizes or browsers it might be slightly less than half, so increasing the thickness may be required.

**Comment by guyhickling** [comment]
Mentioning anti-aliasing just confuses the issue, because dotted outlines are not normally anti-aliased in most browsers - there's no need for it in a straight line. They consist of just a line with one pixel showing the "ink" colour and the next pixel left empty or transparent, and so on alternately.

See also my latest comment in #1369.

**Comment by alastc** [comment]
Hi @guyhickling,

That isn't what I've found, at least when you go above a 1px thickness. See my [test page](https://alastairc.uk/tests/wcag22-examples/focus-more-visible-3.html), there is plenty of anti-aliasing going on for anything 'round', including dotted borders.

Don't forget that displays with more DPI than CSS pixels (e.g. retina displays) can anti-alias even at the 1 CSS pixel size.
